qemu/tests/cfg/qemu_guest_agent.cfg

```diff
@@ -546,30 +546,50 @@
+            cmd_del_key_file = "rm -rf ${guest_homepath}/.ssh/authorized_keys"
+            cmd_install_sshpass = "dnf -y install sshpass"
             variants:
                 - root:
+                    only Linux
                     guest_user = "root"
                     guest_homepath = /${guest_user}
-                    test_login_guest = ssh ${guest_user}@%s -o stricthostkeychecking=no ls ${guest_homepath}
+                    test_login_guest = ssh ${guest_user}@%s -o StrictHostKeyChecking=no ls ${guest_homepath}
                     output_check_str = 'anaconda-ks.cfg'
                 - non_root_user:
+                    only Linux
                     guest_user = "fedora"
-                    guest_user_passwd = "redhat"
                     guest_homepath = "/home/${guest_user}"
-                    cmd_add_user_set_passwd = useradd ${guest_user} && echo ${guest_user_passwd} | passwd --stdin ${guest_user}
+                    cmd_add_user_set_passwd = "useradd ${guest_user} && echo %s | passwd --stdin ${guest_user}"
                     cmd_remove_user = userdel -rf ${guest_user}
-                    test_login_guest = ssh ${guest_user}@%s -o stricthostkeychecking=no ls '/home'
+                    test_login_guest = ssh ${guest_user}@%s -o StrictHostKeyChecking=no ls '/home'
                     output_check_str = '${guest_user}'
-            add_line_at_end = "echo >> ${guest_homepath}/.ssh/authorized_keys"
-            cmd_get_guestkey = "cat ${guest_homepath}/.ssh/authorized_keys"
-            cmd_del_key_file = "rm -rf ${guest_homepath}/.ssh/authorized_keys"
+                - administrator:
+                    only Windows
+                    guest_user = "Administrator"
+                    guest_homepath = "C:\Users\${guest_user}"
+                    cmd_get_guestkey = "powershell.exe Get-Content C:\ProgramData\ssh\administrators_authorized_keys"
+                - non_admin_user:
+                    only Windows
+                    guest_user = "nonadminuser"
+                    guest_homepath = "C:\Users\${guest_user}"
+                    guest_sshdir = "${guest_homepath}\.ssh"
+                    cmd_add_user_set_passwd = "powershell.exe $securePassword = ConvertTo-SecureString -String %s -AsPlainText -Force;New-LocalUser -Name ${guest_user} -Password $securePassword -FullName 'New User' -Description 'Standard non-admin user';Add-LocalGroupMember -Group 'Users' -Member ${guest_user}"
+                    cmd_remove_user = "powershell.exe Remove-LocalUser -Name ${guest_user}"
+                    cmd_get_guestkey = "powershell.exe Get-Content ${guest_homepath}\.ssh\authorized_keys"
+            Windows:
+                install_config_openssh = "powershell.exe Set-ExecutionPolicy -ExecutionPolicy Unrestricted -Scope Process -Force; WIN_UTILS:\Install_config_OpenSSH.ps1"
+                # Please fill out guest_user_passwd before running test
```

Comment:
We need to input passwd before run the test case?I think you can set a password in your internal repo, otherwise if you forget to enter the password, the test case will fail. What do you think?

Reply:
Passwd had already been added to internal repo as previous patch. 
@vivianQizhu  please help to review and merge this patch. thanks.